modules/haplogrep2/classify/main.nf

```diff
@@ -0,0 +1,40 @@
+
+process HAPLOGREP2_CLASSIFY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::haplogrep=2.4.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/haplogrep:2.4.0--hdfd78af_0':
+        'quay.io/biocontainers/haplogrep:2.4.0--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(inputfile)
+    val(format)
+
+    output:
+    tuple val(meta), path("*.txt"), emit: txt
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+
+    """
+    haplogrep \\
+        classify \\
+        $args \\
+        --in $inputfile \\
+        --out ${prefix}.txt \\
+        --format $format
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        haplogrep2: \$(echo \$(haplogrep --version 2>&1) | (sed 's/htt.*//') | (sed 's/.*v//'))
+    END_VERSIONS
```

Comment:
I think the stub section is missing here, and because of it when we run the '_haplogrep2 classify stubs_' test the program creates real outputs. See:

```bash
PROFILE=docker pytest --tag haplogrep2/classify --symlink --kwd --git-aware --color=yes
❯cat /tmp/pytest_workflow_zgsfecec/haplogrep2_classify/output/haplogrep2/test.txt # output directory
"SampleID"      "Haplogroup"    "Rank"  "Quality"       "Range"
"NA12878"       "H2a2a1"        "1"     "0.5000"        "1-16569"
```

Check what we did back in the _haplocheck_ module:

https://github.com/nf-core/modules/blob/c9b29c76869d9713130a13a418c1e8b5aecfb80d/modules/haplocheck/main.nf#L33-L43

Reply:
Thanks! I forgot about adding that part :)